docs/libcudacxx/extended_api/vector_types.rst

```diff
@@ -0,0 +1,132 @@
+.. _libcudacxx-extended-api-vector-types:
+
+Vector Types
+============
+
+The ``<cuda/vector_types>`` header imports the vector types, such as ``float2`` and ``int3``, from the ``<vector_types.h>`` header that comes with the CUDA Toolkit to the ``cuda::`` namespace and provides several additional features to improve their usability.
+
+.. rubric:: Factory Functions
+
+The ``<vector_functions.h>`` header provides factory functions for all of the vector types. In C++, they are practically pointless because we can do `list initialization <https://en.cppreference.com/w/cpp/language/list_initialization>`__, but we provide them for consistency with the CUDA Toolkit and extend them to simplify the creation of the vector types.
+
+For each type (in the example below, only ``float3`` is shown), the following factory functions are provided:
+
+.. code:: cuda
+
+   // equivalent to float3{}
+   [[nodiscard]] __host__ __device__ inline constexpr
+   float3 make_float3() noexcept;
+
+   // equivalent to float3{x, y, z}
+   [[nodiscard]] __host__ __device__ inline constexpr
+   float3 make_float3(float x, float y, float z) noexcept;
+
+   // disambiguation tag for the next overload
+   inline constexpr value_broadcast_t value_broadcast{};
+
+   // equivalent to float3{v, v, v}
+   [[nodiscard]] __host__ __device__ inline constexpr
+   float3 make_float3(value_broadcast_t /* tag */, float v) noexcept;
```

Comment:
Q: What's the problem with `make_float3(float s)` as broadcast? It would be consistent with established GPU languages like OpenCL, GLSL and HLSL.

Reply:
I believe that came from a discussion with me an @davebayer 

I wanted to make sure that it is clear what `make_float3(float s)` does, because there are 2 plausible results:
* {s, 0.0, 0.0}
* {s, s, s}

The `value_broad_cast_t` was a good name that made clear what is happening. Do you have a good name that describes the fill with zero case?

Reply:
> Do you have a good name that describes the fill with zero case?

Yes, aggregate init: `float3{}`.

Reply:
`float3` is an aggregate, so `float3{s, 0, 0}`.